### ISSUE TITLE
predict result is none

### ISSUE THREAD

**Issue by fire717** [iMoonLab/yolov13#18]
I use model s to train and got map0.9+

then I try to predict like this:

```
model = YOLO('runs/detect/train/weights/last.pt') 
result = model.predict(img_path, save=False, imgsz=800, conf=0.25, device='cuda:0')
print(result[0].boxes) #is []
```

I'm sure the test 100 images have object, but all got result [].

**Comment by Mengqi-Lei** [comment]
Thanks for providing detailed code and logs. They make the issue much clearer.

Your situation in brief: the model performs very well during training and validation (mAP > 0.9). However, when you call `predict` separately for inference, the weights and image load successfully but no objects are detected. This is almost certainly because the **parameters used at prediction** or the **data** do not match what was used during training.

Please try the following steps to locate and resolve the problem:

#### 1. Use `best.pt` instead of `last.pt` first

* **`last.pt`** is simply the weights saved at the final training iteration (epoch); the model may already be over-fitting at this point and the performance is not necessarily optimal.
* **`best.pt`** is the weights that achieved the **highest mAP** on the validation set during the entire training process. This is the file you should use for inference and deployment.

#### 2. Check and unify the image size `imgsz`

I noticed you set `imgsz=800` during testing, which is not a common size for YOLO training (640 is typical). You must ensure that the image size (`imgsz`) used in prediction matches the one used in training. Specifically, you can find a file named `args.yaml` in your training output folder `runs/detect/train/`; it records all the parameters used during training, including `imgsz`. Make sure the value you pass to `predict` matches the value in that file.

#### 3. Confirm data-distribution consistency

This is a fundamental check. Ensure that the images you use for `predict` are similar, in visual characteristics, to those used for training and validation. For example, if the model was trained on clear daytime images but you test it on nighttime, blurry, or otherwise very different scenes, performance can drop sharply even if training mAP is high. This is the well-known **domain-shift** problem. Verify whether your test images differ greatly from the training set.

#### 4. Try different confidence thresholds `conf`

If the above steps all check out but the problem persists, try this diagnostic step: temporarily lower `conf` from `0.25` to `0.1` or even lower and see whether any detection boxes appear. This helps determine whether the model is failing to recognise objects entirely or whether the confidences are simply below your current threshold.
